### PR TITLE
Check DNS

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

File: urlextract/urlextract_core.py [hunk 1]
```diff
@@ -65,20 +66,24 @@ class URLExtract(CacheFile):
...
+    def __init__(self, extract_email=False, cache_dns=True, **kwargs):
         """
         Initialize function for URLExtract class.
         Tries to get cached TLDs, if cached file does not exist it will try
         to download new list from IANA and save it to cache file.
 
         :param bool extract_email: True if we want to extract email from text.
             Disabled by default
+        :param bool cache_dns: True replaces socket DNS lookup with caching
+            equivalent provided by velocity or dnspython.
```
Comment (lipoja) [inline urlextract/urlextract_core.py]: Could you please remove `velocity` from this comment since it is not used.

**Comment by lipoja** [comment]
Thanks for this feature and thank you for removing the velocity as dependency. I was going to write you if you could modify the code and do not use it. I will test your PR and then if everything works merge it. 

I really appreciate your time that you spent improving this library! Thanks!

File: urlextract/urlextract_core.py (hunk 1 above)
Comment (jayvdb) [reply urlextract/urlextract_core.py]: Sorry - I had supported velocity until I encountered https://github.com/s0md3v/velocity/issues/2 , but then removed it, and forgot to remove this mention

**Comment by lipoja** [comment]
@jayvdb Do you think that we could agree on not enabling this feature by default? Or at least to discuss it a bit more. 

I assume that there is a lot of use-cases when you want to just extract URLs from text when you do not actually care if the domain is registered or not at the time of extraction. What if there is lot of people counting on this behavior as default. 

When we apply this patch we will change behavior of this library/cli. 
I think this is good feature to have, however I am not sure if we should enforce it by setting it as default.

**Comment by jayvdb** [comment]
I believe that the code currently does not enable it by default
```
PYTHONPATH=. urlextract -c test.txt 
2020-02-11 14:48:29,493 - INFO (filelock): Lock 140699797437968 acquired on /home/jayvdb/projects/python/URLExtract/urlextract/data/tlds-alpha-by-domain.txt.lock
2020-02-11 14:48:29,537 - INFO (filelock): Lock 140699797437968 released on /home/jayvdb/projects/python/URLExtract/urlextract/data/tlds-alpha-by-domain.txt.lock
2020-02-11 14:48:31,255 - INFO (urlextract): Unknown exception during gethostbyname(dfsdfdsdfsf.com) gaierror(-2)
example.com
junk.domain.com
> PYTHONPATH=. urlextract test.txt 
2020-02-11 14:48:38,564 - INFO (filelock): Lock 139759791440616 acquired on /home/jayvdb/projects/python/URLExtract/urlextract/data/tlds-alpha-by-domain.txt.lock
2020-02-11 14:48:38,611 - INFO (filelock): Lock 139759791440616 released on /home/jayvdb/projects/python/URLExtract/urlextract/data/tlds-alpha-by-domain.txt.lock
example.com
junk.domain.com
dfsdfdsdfsf.com
```

`cache_dns` is enabled by default, but it is inactive until `check_dns` is enabled.

**Comment by jayvdb** [comment]
@lipoja , I've split the two features, and added tests for the caching feature.

**Comment by lipoja** [comment]
@jayvdb Thank you for adding the tests and for implementing this feature.

**Pull request merged**